src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -128,116 +303,57 @@ public bool TryNavigateTo(bool usePreviewPane)
             // code must not navigate the selection or caret again if the
             // caret is already within the correct span, otherwise
             // the user cannot navigate the editor anymore.
-            if (this.PersistentSpanValid())
+            if (!this.PersistentSpanValid())
             {
-                if (!SdkUIUtilities.TryGetActiveViewForTextBuffer(this.tag.DocumentPersistentSpan.Span.TextBuffer, out IWpfTextView wpfTextView))
-                {
-                    return false;
-                }
-
-                ITextSnapshot currentSnapshot = this.tag.DocumentPersistentSpan.Span.TextBuffer.CurrentSnapshot;
-
-                // Note that "GetSpan" is not really a great name. What is actually happening
-                // is the "Span" that "GetSpan" is called on is "mapped" onto the passed in
-                // text snapshot. In essence what this means is take the "persistent span"
-                // that we have and "replay" any edits that have occurred and return a new
-                // span. So, if the span is no longer relevant (lets say the text has been deleted)
-                // then you'll get back an empty span.
-                SnapshotSpan trackingSpanSnapshot = this.tag.DocumentPersistentSpan.Span.GetSpan(currentSnapshot);
-
-                // If the caret is already in the text within the marker, don't re-select it
-                // otherwise users cannot move the caret in the region.
-                // If the caret isn't in the marker, move it there.
-                if (!trackingSpanSnapshot.Contains(wpfTextView.Caret.Position.BufferPosition) &&
-                    !trackingSpanSnapshot.IsEmpty)
-                {
-                    wpfTextView.Selection.Select(trackingSpanSnapshot, isReversed: false);
-                    wpfTextView.Caret.MoveTo(trackingSpanSnapshot.End);
-                    wpfTextView.Caret.EnsureVisible();
-                    wpfTextView.VisualElement.Focus();
-                }
-
-                return true;
+                return false;
```

Comment:
throw here

Reply:
There are cases where we may not be able to legitimately create the persistent span (text buffer modified outside of VS, etc).

Throwing is likely not appropriate.

---
In reply to: [501930728](https://github.com/microsoft/sarif-visualstudio-extension/pull/214#discussion_r501930728) [](ancestors = 501930728)